### DISCUSSION TITLE
Кастомизация размеров внутренних частей компонентов

### DISCUSSION THREAD

**Discussion by Yar56** [VKCOM/VKUI#6713] [Q&A] [answered]
допустим у меня  есть компонент <Placeholder/>
у него есть props action
и когда я передаю туда свою разметку я хочу изменить стили обертки для action

на данный момент я делаю это так
задаю плэейсхолдеру className={styles.placeholder}

потом в css 
.placeholder :global(.vkuiPlaceholder__action) {
    width: 100%;
}

т.е. таким образом я цепляюсь за внутреннюю реализацию компонентов
как можно этого избежать?

**Comment by mendrew** [comment, marked as answer]
Привет!

Спасибо за вопрос. 
Действительно, есть такая проблема, с одной стороны мы не советуем завязываться на внутреннюю реализацию компонентов, так как она может поменяться даже в рамках патча, с другой стороны не все компоненты на столько гибкие, чтобы можно было обойтись без этого.

Мы начали работу по переходу к [подкомпонентному](https://github.com/VKCOM/VKUI/issues/4699) подходу, который может позволить, оставаясь в рамках публичного API более гибко настраивать компоненты.

Пример использования такого подхода уже можно посмотреть в документации компонента [Placeholder](https://github.com/VKCOM/VKUI/blob/894f035fa9fc8d109ac9e108ba659b82dd35a106/packages/vkui/src/components/Placeholder/Readme.md#%D0%BF%D0%BE%D0%BA%D0%BE%D0%BC%D0%BF%D0%BE%D0%BD%D0%B5%D0%BD%D1%82%D1%8B).
Вот [ссылка](https://github.com/VKCOM/VKUI/blob/894f035fa9fc8d109ac9e108ba659b82dd35a106/packages/vkui/src/components/Placeholder/Readme.md#%D0%BF%D0%BE%D0%BA%D0%BE%D0%BC%D0%BF%D0%BE%D0%BD%D0%B5%D0%BD%D1%82%D1%8B) на Markdown. 
Доступно с версии [v5.8.0](https://github.com/VKCOM/VKUI/releases/tag/v5.8.0)

Из примера: 
```jsx
<Placeholder.Container>
  <Placeholder.Icon>
    <Icon56UserAddOutline />
  </Placeholder.Icon>
  <Placeholder.Header>Find friends</Placeholder.Header>
  <Placeholder.Text>The people you add as your friends will be displayed here</Placeholder.Text>
  <Placeholder.Actions>
    <ButtonGroup mode="vertical" align="center">
      <Button size="m">Button</Button>
      <Button size="m" mode="tertiary">
        Button
      </Button>
    </ButtonGroup>
  </Placeholder.Actions>
</Placeholder.Container>
```

Тут как раз можно точечно в `Placeholder.Action` передать нужный класс.
```jsx
<Placeholder.Actions className={styles.placeholderAction}>...</PlaceholderAction>
```

К сожалению, ещё не все компоненты умею это, да и не все получится перевести, но базовые мы надеемся в скором времени в этом плане улучшить.

**Reply by Yar56** [reply]
спасибо за ответ)
